app/CocoaPods/CPMetadataTableViewDataSource.swift

```diff
@@ -1,38 +1,79 @@
 import Cocoa
 
+protocol CPMetadataRepresentable: class {
+  var viewIdentifier: String { get }
+  var viewOwner: AnyObject? { get }
+  var rowHeight: CGFloat { get }
+}
+
+extension CPXcodeProject: CPMetadataRepresentable {
+  var viewIdentifier: String { return "xcodeproject_metadata" }
+  var viewOwner: AnyObject? { return self }
+  var rowHeight: CGFloat { return 120 }
+}
+
+extension CPXcodeTarget: CPMetadataRepresentable {
+  var viewIdentifier: String { return "target_metadata" }
+  var viewOwner: AnyObject? { return self }
+  var rowHeight: CGFloat { return 150 }
+}
+
+extension CPPod: CPMetadataRepresentable {
+  var viewIdentifier: String { return "pod_metadata" }
+  var viewOwner: AnyObject? { return self }
+  var rowHeight: CGFloat { return 30 }
+}
+
+class Spacer: NSObject, CPMetadataRepresentable {
+  var viewIdentifier: String { return "spacer" }
+  var viewOwner: AnyObject? { return nil }
+  var rowHeight: CGFloat { return 30 }
+}
+
+protocol CPMetadataExtractable {
+  func metadataForTargets(targets: [CPCocoaPodsTarget]) -> [CPMetadataRepresentable]
+}
+
+extension CPXcodeProject: CPMetadataExtractable {
+  func metadataForTargets(targets: [CPCocoaPodsTarget]) -> [CPMetadataRepresentable] {
+    return [self] + self.targets.flatMap { $0.metadataForTargets(targets) }
+  }
+}
+
+extension CPXcodeTarget: CPMetadataExtractable {
+  func metadataForTargets(targets: [CPCocoaPodsTarget]) -> [CPMetadataRepresentable] {
+    let pods = cocoapodsTargets.flatMap { targetName -> [CPMetadataRepresentable] in
+      targets.filter { $0.name == targetName }
+        .flatMap { $0.pods }
+        .map { $0 as CPMetadataRepresentable }
+    }
+    
+    return [self as CPMetadataRepresentable] + pods + [Spacer() as CPMetadataRepresentable]
```

Comment:
Any idea if this has bad compiler performance characteristics? http://irace.me/swift-profiling/